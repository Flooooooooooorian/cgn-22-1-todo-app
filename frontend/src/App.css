
.page-layout {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    display: grid;
    grid-template-rows: min-content min-content 1fr min-content;
}