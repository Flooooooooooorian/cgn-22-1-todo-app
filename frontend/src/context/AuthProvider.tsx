import {createContext, ReactElement, useEffect, useState} from "react";
import axios from "axios";
import {toast} from "react-toastify";
import {useNavigate} from "react-router-dom";

const LOCALSTORAGE_AUTH_KEY = "todoapptoken123"

export const AuthContext = createContext<{ token: string | undefined, login: (credentials: { username: string, password: string }) => void }>({
    token: undefined,
    login: () => {
        toast.error("Login not initialised")
    }
})

export type AuthProviderProps = {
    children: ReactElement<any, any>
}

export default function AuthProvider({children}: AuthProviderProps) {
    const [token, setToken] = useState<string | undefined>(localStorage.getItem(LOCALSTORAGE_AUTH_KEY) ?? undefined)
    const navigate = useNavigate();

    const login = (credentials: { username: string, password: string }) => {
        axios.post('/auth/login', credentials)
            .then(response => response.data)
            .then((token) => {
                setToken(token)
                return token
            })
            .then(token => localStorage.setItem(LOCALSTORAGE_AUTH_KEY, token))
            .then(() => navigate("/"))
            .catch(() => toast.error("Credentials invalid"))
    }

    return (
        <AuthContext.Provider value={{token, login}}>
            {children}
        </AuthContext.Provider>
    )
}
