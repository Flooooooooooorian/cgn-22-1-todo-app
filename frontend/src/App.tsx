import Header from './components/Header'

import NavigationBar from './components/NavigationBar'
import {BrowserRouter, Route, Routes} from 'react-router-dom'
import Homepage from './components/Homepage'
import DetailsPage from './components/DetailsPage'
import useTodos from './hooks/useTodos'
import './App.css'

function App() {
    const {todos, addTodo, advanceTodo, removeTodo} = useTodos()

    return (
        <BrowserRouter>
            <div className={"page-layout"}>
                <Header/>
                <NavigationBar/>
                <Routes>
                    <Route path="/" element={<Homepage
                        todos={todos}
                        onAdvance={advanceTodo}
                        onDelete={removeTodo}
                        onAdd={addTodo}
                        />}
                    />
                    <Route path={'/todo/:id'} element={<DetailsPage/>}/>
                </Routes>
            </div>
        </BrowserRouter>
    )
}

export default App
