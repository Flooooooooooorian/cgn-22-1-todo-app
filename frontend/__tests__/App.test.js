import renderer from 'react-test-renderer';
import App from "../src/App";
import AuthProvider from "../src/context/AuthProvider";
import {BrowserRouter} from "react-router-dom";

it('renders correctly', () => {
    const app = renderer.create(
        <BrowserRouter>
            <AuthProvider>
                <App/>
            </AuthProvider>
        </BrowserRouter>)
        .toJSON();
    expect(app).toMatchSnapshot();
});
