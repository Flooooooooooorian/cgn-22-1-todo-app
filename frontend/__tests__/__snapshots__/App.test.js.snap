// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="Toastify"
  />,
  <div
    className="page-layout"
  >
    <header>
      <h1>
        Super Todo App ðŸ“—
      </h1>
    </header>
    <div
      className="navigation-bar"
    >
      <a
        aria-current="page"
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </div>
    <main>
      <section
        className="board"
      >
        <div>
          <h2
            className="board-title"
          >
            Open
          </h2>
        </div>
      </section>
      <section
        className="board"
      >
        <div>
          <h2
            className="board-title"
          >
            In Progress
          </h2>
        </div>
      </section>
      <section
        className="board"
      >
        <div>
          <h2
            className="board-title"
          >
            Done
          </h2>
        </div>
      </section>
    </main>
    <form
      className="add-form"
      onSubmit={[Function]}
    >
      <input
        className="description-input"
        onChange={[Function]}
        placeholder="Add a description..."
        value=""
      />
      <button
        className="add-button"
        type="submit"
      >
         Add 
      </button>
    </form>
  </div>,
]
`;
